unity-renderer/Assets/DCLPlugins/ECS7/Systems/VideoPlayerSystem/Tests/ECSVideoPlayerShould.cs

```diff
@@ -0,0 +1,109 @@
+using DCL;
+using DCL.Components.Video.Plugin;
+using System;
+using System.Collections.Generic;
+using DCL.ECS7.InternalComponents;
+using DCL.ECSRuntime;
+using DCL.Shaders;
+using ECSSystems.VideoPlayerSystem;
+using NUnit.Framework;
+using System.Collections;
+using UnityEngine;
+using UnityEngine.TestTools;
+
+namespace Tests
+{
+    public class ECSVideoPlayerSystemShould
+    {
+        private ECS7TestScene scene0;
+        private ECS7TestScene scene1;
+        private InternalECSComponents internalEcsComponents;
+        private ECS7TestUtilsScenesAndEntities testUtils;
+        private Action systemsUpdate;
+        private WebVideoPlayer videoPlayer;
+
+        [SetUp]
+        public void SetUp()
+        {
+            var componentsFactory = new ECSComponentsFactory();
+            var componentsManager = new ECSComponentsManager(componentsFactory.componentBuilders);
+
+            internalEcsComponents = new InternalECSComponents(componentsManager, componentsFactory);
+
+            var videoPlayerSystem = new ECSVideoPlayerSystem(internalEcsComponents.videoPlayerComponent,
+                internalEcsComponents.videoMaterialComponent);
+
+            systemsUpdate = () =>
+            {
+                videoPlayerSystem.Update();
+                internalEcsComponents.WriteSystemUpdate();
+            };
+
+            testUtils = new ECS7TestUtilsScenesAndEntities(componentsManager);
+            scene0 = testUtils.CreateScene(666);
+            scene1 = testUtils.CreateScene(678);
+
+            videoPlayer = new WebVideoPlayer("test", "test.mp4", true, new VideoPluginWrapper_Mock());
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+            testUtils?.Dispose();
+            videoPlayer?.Dispose();
+            AssetPromiseKeeper_Material.i.Cleanup();
+            AssetPromiseKeeper_Texture.i.Cleanup();
+        }
+
+        [UnityTest]
+        public IEnumerator ApplyVideoMaterial()
+        {
+            var playerComponent = internalEcsComponents.videoPlayerComponent;
+            var videoMaterialComponent = internalEcsComponents.videoMaterialComponent;
+
+            ECS7TestEntity entity00 = scene0.CreateEntity(100);
+            ECS7TestEntity entity01 = scene0.CreateEntity(101);
+            ECS7TestEntity entity10 = scene1.CreateEntity(200);
+
+            playerComponent.PutFor(scene0, entity00, new InternalVideoPlayer()
+            {
+                videoPlayer = videoPlayer,
+                assignedMaterials = new List<InternalVideoPlayer.MaterialAssigned>(),
+            });
+
+            videoMaterialComponent.PutFor(scene0, entity00, new InternalVideoMaterial()
+            {
+                material = new Material(Shader.Find("Standard")) { mainTexture = new Texture2D(1, 1), },
```

Comment:
You should be using `Universal Render Pipeline/Lit` since standard is somewhat deprecated a long time ago.